addons/stock/models/stock_rule.py

```diff
@@ -15,6 +15,11 @@
 import logging
 _logger = logging.getLogger(__name__)
 
+class ProcurementException(UserError):
+    def __init__(self, msg, separate_error_messages, procurements):
+        super(UserError, self).__init__(msg)
+        self.separate_error_messages = separate_error_messages
+        self.procurements = procurements
```

Comment:
i would prefer if procurementexcecption does not inherit usererror
it should be a customerror only used in as `run` control flow
``` python
class ProcurementException(Exception):
    def __init__(self, message, procurement):
        Exception.__init__(message)
        self.procurement = procurement
```
an run should convert these procurement exception to an UserError

then in the next commit you can add a kwarg or/and context key to make run not raise the UserError but return you the failed procurements, from which you can get the orderpoint to remove from the next batch